From d91839e6e3c1dbc4e54b48cb5ff96a53675aec6c Mon Sep 17 00:00:00 2001
From: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Date: Wed, 28 Aug 2013 15:32:25 +0700
Subject: Dalvik: Fix Klocwork issue in dvmCompilerPcgGetSymbolAddress

BZ: 134121

The PersistenceInfo->getSymbol does not abort execution if
failOnNotFound parameter is not set to true. Here we do not
expect NULL from getSymbol, thus we are requesting to fail
if symbol not found.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I339d9c566f51e888301483a4f68006b2ff73375f
Orig-MCG-Change-Id: Ia9ecd1ab73afc5cbea3bdfa65df6505d186c2bef
Signed-off-by: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Reviewed-on: http://android.intel.com:8080/128460
Reviewed-by: Katkov, Serguei I <serguei.i.katkov@intel.com>
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Provodin, Vitaly A <vitaly.a.provodin@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/Labels.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/Labels.cpp b/vm/compiler/codegen/x86/pcg/Labels.cpp
index 61f888a..a05c6f7 100644
--- a/vm/compiler/codegen/x86/pcg/Labels.cpp
+++ b/vm/compiler/codegen/x86/pcg/Labels.cpp
@@ -145,7 +145,7 @@ void dvmCompilerPcgBindSymbolAddress (CGSymbol cgSymbol, void *address)
 void* dvmCompilerPcgGetSymbolAddress (CGSymbol cgSymbol)
 {
     //Get the right client symbol information
-    SClientSymbolInfo *clientSymbol = singletonPtr<PersistentInfo> ()->getSymbolInfo (cgSymbol);
+    SClientSymbolInfo *clientSymbol = singletonPtr<PersistentInfo> ()->getSymbolInfo (cgSymbol, true);
 
     //Paranoid
     assert (clientSymbol != 0);
-- 
1.7.4.1

